python/tvm/relay/frontend/tensorflow.py

```diff
@@ -1070,9 +1071,13 @@ def _impl(inputs, attr, params):
 def _range():
     def _impl(inputs, attr, params):
         start = _get_param(params, inputs[0])[0]
-        limit = _get_param(params, inputs[1])[0] \
-            if hasattr(inputs[1], "name_hint") or isinstance(inputs[1], _expr.Constant) \
-            else params.pop('Rank').asnumpy()[0]
+        if hasattr(inputs[1], "name_hint") or isinstance(inputs[1], _expr.Constant):
+            limit = _get_param(params, inputs[1])[0]
+        else:
+            if any(['Rank' in param for param in params]):
+                limit = params.pop('Rank').asnumpy()[0]
+            else:
+                limit = _infer_value_simulated(inputs[1], params).asnumpy()[0]
```

Comment:
Thanks a lot for adding this. While you're here, would you mind adding the fix to `_transpose()` too?

```
...
except (IndexError, KeyError):
    axes = _infer_value_simulated(inputs[1], params)
```

Reply:
@soiferj How can I reproduce the failure? I can add it if I can add a test case for it.

Reply:
I think it should reproduce when the `perm` input is a variable, instead of a constant. For a test case, I suppose you could make it a `placeholder`.

Reply:
https://github.com/apache/incubator-tvm/blob/master/tests/python/frontend/tensorflow/test_forward.py#L2083

Reply:
@soiferj Thank you! I thought about it a bit, it doesn't seem to relate to my pr though the fix looks similar. We can definitely send another pr to fix later. I'd suggest we keep the pr single purpose. What do you think?

Reply:
Sure, that's fine :)